### PR TITLE
Improve the documentation of SubdomainToLevelTranslation.

### PR DESCRIPTION
**Pull request by drwells**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: ibtk/include/ibtk/FEDataManager.h [hunk 1]
```diff
@@ -274,6 +274,18 @@ class FEData : public SAMRAI::tbox::Serializable
 class SubdomainToLevelTranslation
 {
 public:
+    /*!
+     * Constructor. Takes as argument the maximum level number and an input
```
Comment (boyceg) [inline ibtk/include/ibtk/FEDataManager.h:278]:
```suggestion
     * Constructor. Takes as argument the maximum level number in the Cartesian grid patch hierarchy and an input
```

File: ibtk/include/ibtk/FEDataManager.h [hunk 2]
```diff
@@ -283,8 +295,8 @@ class SubdomainToLevelTranslation
     static constexpr int fixed_array_size = 256;
 
     /*!
-     * Return a constant reference. Returns d_max_level_number if no level
-     * is set
+     * Given a libMesh subdomain id, return the patch level on which that
```
Comment (boyceg) [inline ibtk/include/ibtk/FEDataManager.h:298]:
```suggestion
     * Given a libMesh subdomain id, return the patch level of the Cartesian grid hierarchy on which that
```

**Review by boyceg — APPROVED** [review]
Looks good. I think that libMesh may have a notion of level also, and so it might be helpful to specify "level of the Cartesian grid hierarchy" or maybe "level number within the Cartesian grid hierarchy" to be crystal clear.

**Comment by drwells** [comment]
Oh yeah, we even support multilevel FE now so we should fix that.

**Comment by drwells** [comment]
Fixed!

**Review by boyceg — APPROVED** [review]
Looks good to me!

**Comment by boyceg** [comment]
BTW, this is outside of the scope of this PR, but thinking about how "level" is an overloaded term, I wonder if it would be more clear to make the class name `SubdomainToPatchLevelTranslation`.

**Comment by drwells** [comment]
That's easy enough to fix here - lets just rename it while we are at it.

**Comment by drwells** [comment]
I also did another pass on documentation improvements.

**Pull request merged**